src/interop/js-interop/js-types.md

```diff
@@ -0,0 +1,189 @@
+---
+title: JS Types
+description: Usage information about the core types in JS interop.
+---
+
+## Motivation
+
+Dart values and JS values belong to separate domains. When compiling to Wasm,
+they belong in separate *runtimes* as well. As such, users should treat JS
```

Comment:
Hmm, good points. I think we want the user to treat them as foreign types. Dart may or may not treat them as foreign types depending on the compiler, so we want users to play it safe and always treat them as foreign types. I updated some of the wording in other spots too so that it's a bit more clear.

Reply:
Interesting! Ok that makes sense, don't introduce the idea that they could do otherwise. Resolving this one